utils/python/CIME/SystemTests/system_tests_common.py

```diff
@@ -40,45 +41,49 @@ def __init__(self, caseroot=os.getcwd(), case=None):
 
         self._case.set_initial_test_values()
         case_setup(self._caseroot, reset=True, test_mode=True)
-
+        self._case.set_value("TEST",True)
 
     def build(self, sharedlib_only=False, model_only=False):
         build.case_build(self._caseroot, case=self._case,
                          sharedlib_only=sharedlib_only, model_only=model_only)
 
 
     def run(self):
-        with open("TestStatus", 'r') as f:
-            teststatusfile = f.read()
-
         rc, out, err = run_cmd("./case.run", ok_to_fail=True)
         if rc == 0 and self.coupler_log_indicates_run_complete():
-            result = "PASS"
+            self._runstatus = "PASS"
         else:
-            result = "FAIL"
-
-        with open("TestStatus.log", 'a') as f:
-            f.write("case.run output is %s"%out)
-            f.write("case.run error is %s"%err)
-
-        li = teststatusfile.rsplit('PEND', 1)
-        teststatusfile = result.join(li)
-        with open("TestStatus", 'w') as f:
-            f.write(teststatusfile)
+            self._runstatus = "FAIL"
+        if out:
+            appendStatus("case.run output is:\n %s\n"%out, sfile="TestStatus.log")
+        if err:
+            appendStatus("case.run error is:\n %s\n"%err, sfile="TestStatus.log")
+
+    def __del__(self):
```

Comment:
Seems like a good solution. There's a pretty wide timeframe of when this might be called, but it should be guaranteed to be called before the process exits.